test/src/serviceEndToEnd.test.ts

```diff
@@ -369,10 +370,11 @@ describe("Service end-to-end", () => {
             1000000,
             100,
             channelContract.address,
-            KitsuneTools.topics(),
+            KitsuneTools.topics()
         );
 
-        const res = await pisaClient.getAppointmentsByCustomer(account0);
+        const getBlockNumber = await provider.getBlockNumber();
+        const res = await pisaClient.getAppointmentsByCustomer(digest => wallet0.signMessage(arrayify(digest)), account0, getBlockNumber);
         expect(res).to.deep.equal([appointment2.appointment, appointment3.appointment]);
     }).timeout(3000);
 });
```

Comment:
Do you think we should have some tests with failing authentication (missing headers, wrong signature, block more than 5 apart from current head, etc.)?

Reply:
Yeah, we should. Ideally this would be dont with a unit test though, rather than a end to end. The whole service layer lacks unit tests - I have a ticket her: https://github.com/PISAresearch/pisa/issues/352 and I've added these tests as a bullet point there

Reply:
I've also added some more end to end tests on this branch:
- block too low
- block too high
- wrong account
- invalid sig